src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php

```diff
@@ -33,10 +34,40 @@ class AssetsInstallCommand extends ContainerAwareCommand
     const METHOD_ABSOLUTE_SYMLINK = 'absolute symlink';
     const METHOD_RELATIVE_SYMLINK = 'relative symlink';
 
+    private $baseDir;
+    private $filesystem;
+
     /**
-     * @var Filesystem
+     * @param string          $baseDir
+     * @param Filesystem|null $filesystem
      */
-    private $filesystem;
+    public function __construct($baseDir = null, Filesystem $filesystem = null /**, $newApi = false*/)
```

Comment:
> but we never introduced them commented.

> So the argument must exist, and should be documented.

This is a reference i found in master; https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1417

So commented and documented seems fine.

\>=3.4 `__construct('command:name')` or `__construct('/base/path', null, true)`
\>=4.0 `__construct('/base/path')`

We should really prefer base path as a first arg.

I tend to agree with @ogizanagi and be pragmatic

> Simply removing it in 4.0 is enough and won't cause any BC issue